### PR TITLE
Refine SIMD and add test cases for it.

### PR DESCRIPTION
**Pull request by marehr**
Enables SIMD for all platforms and fixes a lot of issues, i.e. wrong SIMD code, not working platforms, and #2038.

### CODE REVIEW

**Comment by h-2** [comment]
Can you provide an overview of the changes? What is all the cmake stuff about and how does this work out if the library is installed and only `seqan-config.cmake` is available?
Is the dependency on Umesimd a hard requirement or an optional dependency?

**Comment by marehr** [comment]
> Can you provide an overview of the changes?

Of course :D

> What is all the cmake stuff about and how does this work out if the library is installed and only `seqan-config.cmake` is available?

All the cmake files are primarily for our build system:
* `FindSDE` utility to find the [Intel Software Emulator](https://software.intel.com/en-us/articles/intel-software-development-emulator), which is used to emulate different target architectures on the same maschine
   
   *NOTE*: This depends only on official cmake scripts.  
*  `FindUmesimd` utility to find the [`UME::SIMD` library](https://github.com/edanor/umesimd)

   *NOTE*: This depends only on official cmake scripts.  
* `SeqAnSimdUtility` gives you `SEQAN_SIMD_(SSE4|AVX2|AVX512_KNL|...)_OPTIONS`, which gives you the compile flags to compile for this architecture and `add_simd_platform_tests(base_target_name)` where you give a base target, which will be cloned and will add the compiler flags for the different architectures using `SEQAN_SIMD_(SSE4|AVX2|AVX512_KNL|...)_OPTIONS`.

  Depending on compiler versions/platform/UME::SIMD, this will disable certain builds and certain tests.

  *NOTE*: This requires `FindSeqan` and optionally `FindUmesimd` and `FindSDE`, but it makes sense to include both.

Thus, all those files could be included in a different projects with only `FindSeqan`.

> Is the dependency on Umesimd a hard requirement or an optional dependency?

Optional, if not present it will not compile the test cases for `UME::SIMD`.

File: include/seqan/align/dp_align_simd_helper.h [hunk 1]
```diff
@@ -89,11 +89,11 @@ struct SimdAlignVariableLengthTraits
...
-#define SEQAN_CREATE_SIMD_REP_IMPL_16(data, strPos, chrPos)   SEQAN_CREATE_SIMD_REP_IMPL_8(data, strPos + 8, chrPos),  SEQAN_CREATE_SIMD_REP_IMPL_8(data, strPos, chrPos)
-#define SEQAN_CREATE_SIMD_REP_IMPL_32(data, strPos, chrPos)   SEQAN_CREATE_SIMD_REP_IMPL_16(data, strPos + 16, chrPos), SEQAN_CREATE_SIMD_REP_IMPL_16(data, strPos, chrPos)
+#define SEQAN_CREATE_SIMD_REP_IMPL_2(data, strPos, chrPos)    getValue(data[strPos], chrPos), getValue(data[strPos + 1], chrPos)
+#define SEQAN_CREATE_SIMD_REP_IMPL_4(data, strPos, chrPos)    SEQAN_CREATE_SIMD_REP_IMPL_2(data, strPos, chrPos),  SEQAN_CREATE_SIMD_REP_IMPL_2(data, strPos + 2, chrPos)
+#define SEQAN_CREATE_SIMD_REP_IMPL_8(data, strPos, chrPos)    SEQAN_CREATE_SIMD_REP_IMPL_4(data, strPos, chrPos),  SEQAN_CREATE_SIMD_REP_IMPL_4(data, strPos + 4, chrPos)
+#define SEQAN_CREATE_SIMD_REP_IMPL_16(data, strPos, chrPos)   SEQAN_CREATE_SIMD_REP_IMPL_8(data, strPos, chrPos),  SEQAN_CREATE_SIMD_REP_IMPL_8(data, strPos + 8, chrPos)
+#define SEQAN_CREATE_SIMD_REP_IMPL_32(data, strPos, chrPos)   SEQAN_CREATE_SIMD_REP_IMPL_16(data, strPos, chrPos), SEQAN_CREATE_SIMD_REP_IMPL_16(data, strPos + 16, chrPos)
```
Comment (marehr) [inline include/seqan/align/dp_align_simd_helper.h]: done

File: include/seqan/align/dp_scout_simd.h [hunk 2]
```diff
@@ -107,10 +110,10 @@ class DPScoutState_<SimdAlignVariableLength<TTraits> >
 
     inline void updateMasksBottom()
     {
-        for (auto pos : sortedEndsV)
+        for (auto posIt  = begin(sortedEndsV, Standard()); posIt != end(sortedEndsV, Standard()); ++posIt)
```
Comment (marehr) [inline include/seqan/align/dp_scout_simd.h]: done

File: include/seqan/basic/debug_test_system.h [hunk 3]
```diff
@@ -1734,7 +1734,7 @@ inline void fail()
 #define SEQAN_SKIP_TEST                         \
     do {                                        \
         ::seqan::ClassTest::skipCurrentTest();  \
-        return;                                 \
+        throw ::seqan::ClassTest::AssertionFailedException(); \
```
Comment (marehr) [inline include/seqan/basic/debug_test_system.h]: done

File: include/seqan/simd.h [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+#endif
+
+// Seqan's simd implementation supports only avx2 and sse4.
+#if defined(SEQAN_SEQANSIMD_ENABLED) && !(defined(__AVX2__) || (defined(__SSE4_1__) && defined(__SSE4_2__)))
+    #undef SEQAN_SIMD_ENABLED
+    #undef SEQAN_SEQANSIMD_ENABLED
+#endif
+
+#if defined(SEQAN_SEQANSIMD_ENABLED) && (defined(COMPILER_MSVC) || defined(COMPILER_WINTEL))
+    #error SEQAN::SIMD (vector extension) is not supported by msvc and windows intel compiler
```
Comment (marehr) [inline include/seqan/simd.h]: done

File: include/seqan/simd.h [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+#endif
+
+// Define maximal size of vector in byte.
+#if !defined(SEQAN_SEQANSIMD_ENABLED) && defined(__AVX512F__)
+    // SEQAN_SIMD doesn't support AVX512, thus fallback to AVX2 (One can assume
```
Comment (marehr) [inline include/seqan/simd.h]: done

File: include/seqan/simd/simd_base.h [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+struct Value<SimdVector<TValue, LENGTH> >
+{
+    typedef TValue Type;
+};
+
+template <typename TValue, int LENGTH>
+struct Value<SimdVector<TValue, LENGTH> const>:
+    public Value<SimdVector<TValue, LENGTH> >
+{
+};
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: done

File: include/seqan/simd/simd_base.h [hunk 7]
```diff
@@ -0,0 +1,322 @@
...
+struct LENGTH<SimdVector<TValue, LENGTH_> >
+{
+    enum { VALUE = LENGTH_ };
+};
+
+template <typename TValue, int LENGTH_>
+struct LENGTH<SimdVector<TValue, LENGTH_> const>:
+    public LENGTH<SimdVector<TValue, LENGTH_> >
+{
+};
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: done

File: include/seqan/simd/simd_base.h [hunk 8]
```diff
@@ -0,0 +1,322 @@
...
+ * TSimdVector vec {2, 4, 8, 16}, res;
+ * TSimdSwizzleVector swizzle {3, 2, 0, 2};
+ *
+ * res = shuffleVector(vec, swizzle); // res = {16, 8, 2, 8}
+ * ```
+ */
+template <typename TSimdVector>
+struct SimdSwizzleVector : SimdSwizzleVectorImpl<TSimdVector, typename Is<SimdVectorConcept<TSimdVector> >::Type >
+{
+};
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: done

File: include/seqan/simd/simd_base.h [hunk 9]
```diff
@@ -0,0 +1,322 @@
...
+ * getValue(a, pos);
+ *
+ * // same as
+ *
+ * a[pos];
+ * ```
+ */
+template <typename TSimdVector, typename TPosition>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector> >, typename Value<TSimdVector>::Type)
+getValue(TSimdVector const &vector, TPosition const pos);
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: done

File: include/seqan/simd/simd_base.h [hunk 10]
```diff
@@ -0,0 +1,322 @@
...
+ * assignValue(a, pos, value);
+ *
+ * // same as
+ *
+ * a[pos] = value;
+ * ```
+ */
+template <typename TSimdVector, typename TPosition, typename TValue2>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector> >, void)
+assignValue(TSimdVector &vector, TPosition const pos, TValue2 const value);
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: done

File: include/seqan/simd/simd_base_seqan_impl_avx2.h [hunk 11]
```diff
@@ -0,0 +1,1352 @@
...
+            ),
+            _mm256_set1_epi32(mask)
+        )
+    );
+}
+
+template <typename TValue, typename TSize, TSize SCALE>
+inline __m256i
+seqan_mm256_i32gather_epi(TValue const * memAddr,
+                            __m256i const & idx,
```
Comment (rrahn) [inline include/seqan/simd/simd_base_seqan_impl_avx2.h]: alignment of arguments

File: include/seqan/simd/simd_base_seqan_impl_avx2.h [hunk 12]
```diff
@@ -0,0 +1,1352 @@
...
+    // we need a look-up table to reverse the lowest 4 bits
+    // in order to place the permute the transposed rows
+    static const unsigned char bitRev[] = { 0, 8, 4,12, 2,10, 6,14, 1, 9, 5,13, 3,11, 7,15,
+                                           16,24,20,28,18,26,22,30,17,25,21,29,19,27,23,31};
+
+    // transpose a 32x32 byte matrix
+    __m256i tmp1[32];
+    for (int i = 0; i < 16; ++i)
+    {
+        tmp1[i]    = _mm256_unpacklo_epi8(SEQAN_VECTOR_CAST_(const __m256i &, matrix[2*i]), SEQAN_VECTOR_CAST_(const __m256i &, matrix[2*i+1]));
```
Comment (rrahn) [inline include/seqan/simd/simd_base_seqan_impl_avx2.h]: wrap lines

File: include/seqan/simd/simd_base.h [hunk 13]
```diff
@@ -0,0 +1,322 @@
...
+
+template <int ROWS, typename TSimdVector>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector> >, void)
+transpose(TSimdVector matrix[ROWS]);
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: we decided, that this is not important :grin:

File: include/seqan/simd/simd_base.h [hunk 14]
```diff
@@ -0,0 +1,322 @@
...
+ *
+ * // same as
+ *
+ * for(auto i = 0u; i < LENGTH; ++i)
+ *     c[i] = 0;
+ * ```
+ */
+template <typename TSimdVector>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector> >, void)
+clearVector(TSimdVector &vector);
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: done

File: include/seqan/simd/simd_base_seqan_impl_sse4.2.h [hunk 15]
```diff
@@ -0,0 +1,892 @@
...
+// ============================================================================
+// Functions
+// ============================================================================
+
+// --------------------------------------------------------------------------
+// _fillVector (128bit)
+// --------------------------------------------------------------------------
+
+template <typename TSimdVector, typename... TValue>
+inline void _fillVector(TSimdVector &vector, std::tuple<TValue...> const & x, std::index_sequence<0> const &, SimdParams_<16, 16>) { vector = SEQAN_VECTOR_CAST_(TSimdVector, _mm_set1_epi8(std::get<0>(x))); }
```
Comment (rrahn) [inline include/seqan/simd/simd_base_seqan_impl_sse4.2.h]: wrap lines

File: include/seqan/simd/simd_base_seqan_impl_sse4.2.h [hunk 16]
```diff
@@ -0,0 +1,892 @@
...
+
+template <typename TSimdVector, int L>
+inline void _clearVector(TSimdVector &vector, SimdParams_<16, L>) { vector = SEQAN_VECTOR_CAST_(TSimdVector, _mm_setzero_si128()); }
+
+// --------------------------------------------------------------------------
+// _createVector (128bit)
+// --------------------------------------------------------------------------
+
+template <typename TSimdVector, typename TValue>
+inline TSimdVector _createVector(TValue x, SimdParams_<16, 16>) { return SEQAN_VECTOR_CAST_(TSimdVector, _mm_set1_epi8(x)); }
```
Comment (rrahn) [inline include/seqan/simd/simd_base_seqan_impl_sse4.2.h]: wrap lines

File: include/seqan/simd/simd_base_seqan_impl_sse4.2.h [hunk 17]
```diff
@@ -0,0 +1,892 @@
...
+                              _mm_cmpeq_epi64(SEQAN_VECTOR_CAST_(const __m128i&, a),
+                                              _mm_setzero_si128()));
+}
+
+// --------------------------------------------------------------------------
+// _divide (128bit)
+// --------------------------------------------------------------------------
+
+template <typename TSimdVector>
+inline TSimdVector _divide(TSimdVector &a, int b, SimdParams_<16, 16>) { return SEQAN_VECTOR_CAST_(TSimdVector, _mm_div_epi8(a, _mm_set1_epi8(b))); }
```
Comment (rrahn) [inline include/seqan/simd/simd_base_seqan_impl_sse4.2.h]: wrap lines

File: include/seqan/simd/simd_base_umesimd_impl.h [hunk 18]
```diff
@@ -0,0 +1,603 @@
...
+getValue(UME::SIMD::SIMDSwizzle<LENGTH> const &vector, TPosition const pos)
+{
+    return vector[pos];
+}
+
+template <uint32_t LENGTH, typename TPosition>
+inline typename Value<UME::SIMD::SIMDSwizzle<LENGTH> >::Type
+value(UME::SIMD::SIMDSwizzle<LENGTH> const &vector, TPosition const pos)
+{
+
```
Comment (rrahn) [inline include/seqan/simd/simd_base_umesimd_impl.h]: empty line

File: include/seqan/simd/simd_base_umesimd_impl.h [hunk 19]
```diff
@@ -0,0 +1,603 @@
...
+SEQAN_CONCEPT_IMPL((typename UME::SIMD::SIMDSwizzle<LENGTH> const), (SimdVectorConcept));
+
+template <uint32_t LENGTH>
+struct Value<UME::SIMD::SIMDSwizzle<LENGTH> >
+{
+    typedef uint32_t Type;
+};
+
+template <uint32_t LENGTH_>
+struct LENGTH<UME::SIMD::SIMDSwizzle<LENGTH_> > {
```
Comment (rrahn) [inline include/seqan/simd/simd_base_umesimd_impl.h]: brace on new line

File: include/seqan/simd/simd_base.h [hunk 20]
```diff
@@ -0,0 +1,322 @@
...
+ *
+ * a[0] = 1;
+ * a[1] = 3;
+ * a[2] = 13;
+ * a[3] = 1337;
+ * ```
+ */
+template <typename TSimdVector, typename ...TValue>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector> >, void)
+fillVector(TSimdVector &vector, TValue const... args);
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: done

File: include/seqan/simd/simd_base.h [hunk 21]
```diff
@@ -0,0 +1,322 @@
...
+ * c = a == b;
+ * ```
+ */
+template <typename TSimdVector>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector> >, TSimdVector)
+cmpEq (TSimdVector const &a, TSimdVector const &b);
+
+template <typename TSimdVector>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector> >, TSimdVector)
+operator == (TSimdVector const &a, TSimdVector const &b);
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: done

File: include/seqan/simd/simd_base.h [hunk 22]
```diff
@@ -0,0 +1,322 @@
...
+ * c = cmpGt(a, b);
+ *
+ * // same as
+ *
+ * c = a > b;
+ * ```
+ */
+template <typename TSimdVector>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector> >, TSimdVector)
+cmpGt (TSimdVector const &a, TSimdVector const &b);
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: done

File: include/seqan/simd/simd_base.h [hunk 23]
```diff
@@ -0,0 +1,322 @@
...
+
+template <typename TSimdVector>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector> >, TSimdVector)
+operator > (TSimdVector const &a, TSimdVector const &b);
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: done

File: include/seqan/simd/simd_base.h [hunk 24]
```diff
@@ -0,0 +1,322 @@
...
+
+template <typename TSimdVector>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector> >, TSimdVector)
+max(TSimdVector const &a, TSimdVector const &b);
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: done

File: include/seqan/simd/simd_base.h [hunk 25]
```diff
@@ -0,0 +1,322 @@
...
+
+template <typename TSimdVector>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector> >, TSimdVector)
+operator | (TSimdVector const &a, TSimdVector const &b);
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: done

File: include/seqan/simd/simd_base.h [hunk 26]
```diff
@@ -0,0 +1,322 @@
...
+
+template <typename TSimdVector>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector> >, TSimdVector &)
+operator |= (TSimdVector &a, TSimdVector const &b);
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: done

File: include/seqan/simd/simd_base.h [hunk 27]
```diff
@@ -0,0 +1,322 @@
...
+
+template <typename TSimdVector>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector> >, TSimdVector)
+operator & (TSimdVector const &a, TSimdVector const &b);
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: done

File: include/seqan/simd/simd_base.h [hunk 28]
```diff
@@ -0,0 +1,322 @@
...
+
+template <typename TSimdVector>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector> >, TSimdVector &)
+operator &= (TSimdVector &a, TSimdVector const &b);
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: done

File: include/seqan/simd/simd_base.h [hunk 29]
```diff
@@ -0,0 +1,322 @@
...
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector> >, TSimdVector)
+gather(TValue const * memAddr, TSimdVector const & idx);
+
+template <typename TSimdVector1, typename TSimdVector2>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector1> >, TSimdVector1)
+shuffleVector(TSimdVector1 const &vector, TSimdVector2 const &indices);
+
+// NOTE(rmaerker): Make this function available, also if SIMD is not enabled.
+template <typename TSimdVector, typename TValue>
+inline SEQAN_FUNC_DISABLE_IF(Is<SimdVectorConcept<TSimdVector> >, TSimdVector)
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: should stay the same function definition as in the other cases

File: include/seqan/simd/simd_base.h [hunk 30]
```diff
@@ -0,0 +1,322 @@
...
+ *
+ * // same as
+ *
+ * for(auto i = 0u; i < LENGTH; ++i)
+ *     c[i] = a;
+ * ```
+ */
+template <typename TSimdVector, typename TValue>
+inline SEQAN_FUNC_ENABLE_IF(Is<SimdVectorConcept<TSimdVector> >, TSimdVector)
+createVector(TValue x);
```
Comment (marehr) [inline include/seqan/simd/simd_base.h]: done

File: include/seqan/simd/simd_base_seqan_impl_avx2.h [hunk 31]
```diff
@@ -0,0 +1,1352 @@
...
+    constexpr auto const mask = static_cast<uint64_t>(MaxValue<TUnsignedValue>::VALUE);
+
+    return _mm256_and_si256(
+        _mm256_i64gather_epi64((const long long *) memAddr, idx, SCALE),
+        _mm256_set1_epi64x(mask)
+    );
+}
+
+template <typename TValue, typename TSimdVector, typename TSize, TSize SCALE>
+inline TSimdVector _gather(TValue const * memAddr,
```
Comment (marehr) [inline include/seqan/simd/simd_base_seqan_impl_avx2.h]: done

File: include/seqan/simd/simd_base_seqan_impl_avx2.h [hunk 32]
```diff
@@ -0,0 +1,1352 @@
...
+        seqan_mm256_i8gather_epi(
+            memAddr,
+            SEQAN_VECTOR_CAST_(__m256i const &, idx),
+            scale
+        )
+    );
+}
+
+template <typename TSimdVector, typename TSize, TSize SCALE>
+inline TSimdVector _gather(int8_t const * memAddr,
```
Comment (marehr) [inline include/seqan/simd/simd_base_seqan_impl_avx2.h]: done

File: include/seqan/align/dp_scout_simd.h [hunk 33]
```diff
@@ -233,7 +236,16 @@ _updateHostPositions(DPScout_<TDPCell, TScoutSpec> & dpScout,
...
-#if defined(__AVX2__)
+#if SEQAN_UMESIMD_ENABLED
+    using TSimdHalfVec = typename UME::SIMD::SIMDTraits<TSimdVec>::HALF_LEN_VEC_T;
+    TSimdHalfVec cmpLow, cmpHigh;
+    cmp.unpack(cmpLow, cmpHigh);
+
+    dpScout._maxHostLow = blend(dpScout._maxHostLow, positionNavigator,
+                                static_cast<SimdVector<int32_t>::Type>(cmpLow));
+    dpScout._maxHostHigh = blend(dpScout._maxHostHigh, positionNavigator,
+                                static_cast<SimdVector<int32_t>::Type>(cmpHigh));
```
Comment (marehr) [inline include/seqan/align/dp_scout_simd.h]: done

File: include/seqan/simd/simd_base_seqan_impl_avx2.h [hunk 34]
```diff
@@ -0,0 +1,1352 @@
...
+    return SEQAN_VECTOR_CAST_(TSimdVector,
+                              _mm256_cmpeq_epi64(SEQAN_VECTOR_CAST_(const __m256i&, a), _mm256_setzero_si256()));
+}
+
+// --------------------------------------------------------------------------
+// _divide (256bit)
+// --------------------------------------------------------------------------
+
+template <typename TSimdVector>
+inline TSimdVector _divide(TSimdVector &a, int b, SimdParams_<32, 32>) { return SEQAN_VECTOR_CAST_(TSimdVector, _mm256_div_epi8(a, _mm256_set1_epi8(b))); }
```
Comment (marehr) [inline include/seqan/simd/simd_base_seqan_impl_avx2.h]: done

File: include/seqan/simd/simd_base_seqan_impl_avx2.h (hunk 11 above)
Comment (marehr) [inline include/seqan/simd/simd_base_seqan_impl_avx2.h]: done

File: include/seqan/simd/simd_base_seqan_impl_avx2.h (hunk 12 above)
Comment (marehr) [inline include/seqan/simd/simd_base_seqan_impl_avx2.h]: done

File: include/seqan/simd/simd_base_seqan_impl_sse4.2.h (hunk 15 above)
Comment (marehr) [inline include/seqan/simd/simd_base_seqan_impl_sse4.2.h]: done

File: include/seqan/simd/simd_base_seqan_impl_sse4.2.h (hunk 16 above)
Comment (marehr) [inline include/seqan/simd/simd_base_seqan_impl_sse4.2.h]: done

File: include/seqan/simd/simd_base_seqan_impl_sse4.2.h (hunk 17 above)
Comment (marehr) [inline include/seqan/simd/simd_base_seqan_impl_sse4.2.h]: done

File: include/seqan/simd/simd_base_umesimd_impl.h (hunk 19 above)
Comment (marehr) [inline include/seqan/simd/simd_base_umesimd_impl.h]: done

File: include/seqan/simd/simd_base_umesimd_impl.h (hunk 18 above)
Comment (marehr) [inline include/seqan/simd/simd_base_umesimd_impl.h]: done

**Comment by marehr** [comment]
The builds finally went through, @h-2, @rrahn 
more remarks?

**Comment by rrahn** [comment]
let's ship it! :godmode:

**Pull request merged**